src/runtime/styled.js

```diff
@@ -11,19 +13,55 @@ function omitNonHostProps(props) {
   return result;
 }
 
-export const styled = (type, displayName, styles, getStyles) => {
-  const styleMap = getStyles(styles);
+function styled(type, displayName, styles, camelName, kebabName) {
+  const componentClassName = has.call(styles, camelName)
+    ? styles[camelName]
+    : styles[kebabName];
   const omit = typeof type === 'string' ? omitNonHostProps : null;
 
-  function Styled(_props) {
-    const props = omit ? omit(_props) : { ..._props };
-    delete props.innerRef;
-    props.ref = _props.innerRef;
-    props.className = classNames(
-      props.className,
-      styleMap.map(s => (typeof s === 'string' ? s : s(_props))),
-    );
-    return React.createElement(type, props);
+  const hasModifiers = Object.keys(styles).some(
+    className => className !== camelName && className !== kebabName,
+  );
+
+  function Styled(props) {
+    const childProps = omit ? omit(props) : { ...props };
+
+    delete childProps.innerRef;
+    childProps.ref = props.innerRef;
+
+    if (hasModifiers) {
```

Comment:
probably better to iterate through props rather than iterating through defined styles?